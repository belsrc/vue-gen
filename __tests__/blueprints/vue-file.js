/* eslint-disable fp-jxl/no-unused-expression, fp-jxl/no-nil */
const genVue = require('./../../src/blueprints/vue-file');

const compName = 'TestComponent';
const className = 'test-component';

describe('genVue', () => {
  // Vanilla
  test('has the expected ouput structure not as page and without state', () => {
    const actual = genVue(compName, className, false, false, false);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure as page and with state', () => {
    const actual = genVue(compName, className, true, true, false);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure as page without state', () => {
    const actual = genVue(compName, className, true, false, false);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure not as page with state', () => {
    const actual = genVue(compName, className, false, true, false);

    expect(actual).toMatchSnapshot();
  });

  // Typescript
  test('has the expected ouput structure with Typescript not as page and without state', () => {
    const actual = genVue(compName, className, false, false, true);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure with Typescript as page and with state', () => {
    const actual = genVue(compName, className, true, true, true);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure with Typescript as page without state', () => {
    const actual = genVue(compName, className, true, false, true);

    expect(actual).toMatchSnapshot();
  });

  test('has the expected ouput structure with Typescript not as page with state', () => {
    const actual = genVue(compName, className, false, true, true);

    expect(actual).toMatchSnapshot();
  });
});
