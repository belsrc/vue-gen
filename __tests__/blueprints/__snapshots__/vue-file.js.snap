// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genVue has the expected ouput structure as page and with state 1`] = `
"<script>
import { mapState, mapGetters, mapActions } from 'vuex';

export default {
  name: 'TestComponent',

  components: {},

  props: {},

  data() {
    return {};
  },

  computed: {
    ...mapState({}),

    ...mapGetters({}),
  },

  watch: {},

  async created() {
    await this.loadPageData();
  },

  mounted() {},

  // When route changes and this component is already rendered, this guard is used
  async beforeRouteUpdate(to, from, next) {
    await this.loadPageData();

    return next();
  },

  methods: {
    ...mapActions({}),

    async loadPageData() {},
  },
};
</script>

<template>
  <div class=\\"test-component\\">
    (ಠ益ಠ)
  </div>
</template>

<style lang=\\"scss\\">
  // Mobile / Baseline
  .test-component {}

  // Tablets
  @media only screen and (min-width: 768px) {
    .test-component {}
  }

  // Hi-Res Tablets
  @media only screen and (min-width: 1024px) {
    .test-component {}
  }

  // Desktops
  @media only screen and (min-width: 1200px) {
    .test-component {}
  }

  // Mid DPI Laptop/Desktop
  @media only screen and (min-width: 1360px) {
    .test-component {}
  }

  // Hi DPI Desktop
  @media only screen and (min-width: 1440px) {
    .test-component {}
  }

  // Larger Desktops
  @media only screen and (min-width: 1920px) {
    .test-component {}
  }
</style>
"
`;

exports[`genVue has the expected ouput structure as page without state 1`] = `
"<script>
export default {
  name: 'TestComponent',

  components: {},

  props: {},

  data() {
    return {};
  },

  computed: {},

  watch: {},

  async created() {
    await this.loadPageData();
  },

  mounted() {},

  // When route changes and this component is already rendered, this guard is used
  async beforeRouteUpdate(to, from, next) {
    await this.loadPageData();

    return next();
  },

  methods: {
    async loadPageData() {},
  },
};
</script>

<template>
  <div class=\\"test-component\\">
    (ಠ益ಠ)
  </div>
</template>

<style lang=\\"scss\\">
  // Mobile / Baseline
  .test-component {}

  // Tablets
  @media only screen and (min-width: 768px) {
    .test-component {}
  }

  // Hi-Res Tablets
  @media only screen and (min-width: 1024px) {
    .test-component {}
  }

  // Desktops
  @media only screen and (min-width: 1200px) {
    .test-component {}
  }

  // Mid DPI Laptop/Desktop
  @media only screen and (min-width: 1360px) {
    .test-component {}
  }

  // Hi DPI Desktop
  @media only screen and (min-width: 1440px) {
    .test-component {}
  }

  // Larger Desktops
  @media only screen and (min-width: 1920px) {
    .test-component {}
  }
</style>
"
`;

exports[`genVue has the expected ouput structure not as page and without state 1`] = `
"<script>
export default {
  name: 'TestComponent',

  components: {},

  props: {},

  data() {
    return {};
  },

  computed: {},

  watch: {},

  created() {},

  mounted() {},

  methods: {},
};
</script>

<template>
  <div class=\\"test-component\\">
    (ಠ益ಠ)
  </div>
</template>

<style lang=\\"scss\\">
  // Mobile / Baseline
  .test-component {}

  // Tablets
  @media only screen and (min-width: 768px) {
    .test-component {}
  }

  // Hi-Res Tablets
  @media only screen and (min-width: 1024px) {
    .test-component {}
  }

  // Desktops
  @media only screen and (min-width: 1200px) {
    .test-component {}
  }

  // Mid DPI Laptop/Desktop
  @media only screen and (min-width: 1360px) {
    .test-component {}
  }

  // Hi DPI Desktop
  @media only screen and (min-width: 1440px) {
    .test-component {}
  }

  // Larger Desktops
  @media only screen and (min-width: 1920px) {
    .test-component {}
  }
</style>
"
`;

exports[`genVue has the expected ouput structure not as page with state 1`] = `
"<script>
import { mapState, mapGetters, mapActions } from 'vuex';

export default {
  name: 'TestComponent',

  components: {},

  props: {},

  data() {
    return {};
  },

  computed: {
    ...mapState({}),

    ...mapGetters({}),
  },

  watch: {},

  created() {},

  mounted() {},

  methods: {
    ...mapActions({}),
  },
};
</script>

<template>
  <div class=\\"test-component\\">
    (ಠ益ಠ)
  </div>
</template>

<style lang=\\"scss\\">
  // Mobile / Baseline
  .test-component {}

  // Tablets
  @media only screen and (min-width: 768px) {
    .test-component {}
  }

  // Hi-Res Tablets
  @media only screen and (min-width: 1024px) {
    .test-component {}
  }

  // Desktops
  @media only screen and (min-width: 1200px) {
    .test-component {}
  }

  // Mid DPI Laptop/Desktop
  @media only screen and (min-width: 1360px) {
    .test-component {}
  }

  // Hi DPI Desktop
  @media only screen and (min-width: 1440px) {
    .test-component {}
  }

  // Larger Desktops
  @media only screen and (min-width: 1920px) {
    .test-component {}
  }
</style>
"
`;

exports[`genVue has the expected ouput structure with Typescript as page and with state 1`] = `
"<script lang=\\"ts\\">
import { mapState, mapGetters, mapActions } from 'vuex';

export default {
  name: 'TestComponent',

  components: {},

  props: {},

  data() {
    return {};
  },

  computed: {
    ...mapState({}),

    ...mapGetters({}),
  },

  watch: {},

  async created() {
    await this.loadPageData();
  },

  mounted() {},

  // When route changes and this component is already rendered, this guard is used
  async beforeRouteUpdate(to, from, next) {
    await this.loadPageData();

    return next();
  },

  methods: {
    ...mapActions({}),

    async loadPageData() {},
  },
};
</script>

<template>
  <div class=\\"test-component\\">
    (ಠ益ಠ)
  </div>
</template>

<style lang=\\"scss\\">
  // Mobile / Baseline
  .test-component {}

  // Tablets
  @media only screen and (min-width: 768px) {
    .test-component {}
  }

  // Hi-Res Tablets
  @media only screen and (min-width: 1024px) {
    .test-component {}
  }

  // Desktops
  @media only screen and (min-width: 1200px) {
    .test-component {}
  }

  // Mid DPI Laptop/Desktop
  @media only screen and (min-width: 1360px) {
    .test-component {}
  }

  // Hi DPI Desktop
  @media only screen and (min-width: 1440px) {
    .test-component {}
  }

  // Larger Desktops
  @media only screen and (min-width: 1920px) {
    .test-component {}
  }
</style>
"
`;

exports[`genVue has the expected ouput structure with Typescript as page without state 1`] = `
"<script lang=\\"ts\\">
export default {
  name: 'TestComponent',

  components: {},

  props: {},

  data() {
    return {};
  },

  computed: {},

  watch: {},

  async created() {
    await this.loadPageData();
  },

  mounted() {},

  // When route changes and this component is already rendered, this guard is used
  async beforeRouteUpdate(to, from, next) {
    await this.loadPageData();

    return next();
  },

  methods: {
    async loadPageData() {},
  },
};
</script>

<template>
  <div class=\\"test-component\\">
    (ಠ益ಠ)
  </div>
</template>

<style lang=\\"scss\\">
  // Mobile / Baseline
  .test-component {}

  // Tablets
  @media only screen and (min-width: 768px) {
    .test-component {}
  }

  // Hi-Res Tablets
  @media only screen and (min-width: 1024px) {
    .test-component {}
  }

  // Desktops
  @media only screen and (min-width: 1200px) {
    .test-component {}
  }

  // Mid DPI Laptop/Desktop
  @media only screen and (min-width: 1360px) {
    .test-component {}
  }

  // Hi DPI Desktop
  @media only screen and (min-width: 1440px) {
    .test-component {}
  }

  // Larger Desktops
  @media only screen and (min-width: 1920px) {
    .test-component {}
  }
</style>
"
`;

exports[`genVue has the expected ouput structure with Typescript not as page and without state 1`] = `
"<script lang=\\"ts\\">
export default {
  name: 'TestComponent',

  components: {},

  props: {},

  data() {
    return {};
  },

  computed: {},

  watch: {},

  created() {},

  mounted() {},

  methods: {},
};
</script>

<template>
  <div class=\\"test-component\\">
    (ಠ益ಠ)
  </div>
</template>

<style lang=\\"scss\\">
  // Mobile / Baseline
  .test-component {}

  // Tablets
  @media only screen and (min-width: 768px) {
    .test-component {}
  }

  // Hi-Res Tablets
  @media only screen and (min-width: 1024px) {
    .test-component {}
  }

  // Desktops
  @media only screen and (min-width: 1200px) {
    .test-component {}
  }

  // Mid DPI Laptop/Desktop
  @media only screen and (min-width: 1360px) {
    .test-component {}
  }

  // Hi DPI Desktop
  @media only screen and (min-width: 1440px) {
    .test-component {}
  }

  // Larger Desktops
  @media only screen and (min-width: 1920px) {
    .test-component {}
  }
</style>
"
`;

exports[`genVue has the expected ouput structure with Typescript not as page with state 1`] = `
"<script lang=\\"ts\\">
import { mapState, mapGetters, mapActions } from 'vuex';

export default {
  name: 'TestComponent',

  components: {},

  props: {},

  data() {
    return {};
  },

  computed: {
    ...mapState({}),

    ...mapGetters({}),
  },

  watch: {},

  created() {},

  mounted() {},

  methods: {
    ...mapActions({}),
  },
};
</script>

<template>
  <div class=\\"test-component\\">
    (ಠ益ಠ)
  </div>
</template>

<style lang=\\"scss\\">
  // Mobile / Baseline
  .test-component {}

  // Tablets
  @media only screen and (min-width: 768px) {
    .test-component {}
  }

  // Hi-Res Tablets
  @media only screen and (min-width: 1024px) {
    .test-component {}
  }

  // Desktops
  @media only screen and (min-width: 1200px) {
    .test-component {}
  }

  // Mid DPI Laptop/Desktop
  @media only screen and (min-width: 1360px) {
    .test-component {}
  }

  // Hi DPI Desktop
  @media only screen and (min-width: 1440px) {
    .test-component {}
  }

  // Larger Desktops
  @media only screen and (min-width: 1920px) {
    .test-component {}
  }
</style>
"
`;
